Update the 'pipsupported' version to 25.2

sirosen opened the pull request:
Update the 'pipsupported' build, now that there's a new release.

##### Contributor checklist

- [x] Included tests for the changes.
- [x] A change note is created in `changelog.d/` (see [`changelog.d/README.md`](https://github.com/jazzband/pip-tools/blob/main/changelog.d/#readme) for instructions) or the PR text says "no changelog needed".

##### Maintainer checklist

- [ ] If no changelog is needed, apply the `skip-changelog` label.
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

webknjaz commented:
By the way, there was talk about rephrasing the tox factors. I think @hugovk mentioned it being confusing.. But that would be a separate PR, of course.

webknjaz commented:
Oh.. RTD is still configured with their control.

Could you convert it to use this config instead https://github.com/sphinx-contrib/sphinxcontrib-towncrier/blob/27bdb93/.readthedocs.yaml, in a separate PR?

jayaddison commented on changelog.d/2214.contrib.md:
```diff
@@ -0,0 +1 @@
+`pip-tools` now tests on and officially supports `pip` version 25.2 .
```
NB: I think a byline may be missing here?  (noticed during preparation of #2217)

```suggestion
`pip-tools` now tests on and officially supports `pip` version 25.2 .
 -- by {user}`sirosen`
```

sirosen replied:
I don't mind always including the byline if it helps us keep the style consistent for contributor PRs.

I have a couple of projects in which I give "`Thanks @user`" for 3rd party contributions and not for maintainer-driven changes, which is probably the habit that led to not having one here.

sirosen commented on changelog.d/2214.contrib.md:
```diff
@@ -0,0 +1 @@
+`pip-tools` now tests on and officially supports `pip` version 25.2 .
```
I like the idea (yes, I find it awkward to put punctuation right next to an unadorned version number), but it turned out not to be necessary after adding a byline.

sirosen commented:
I've given this a rebase to try to get things in order for us to do a 7.5.1 release.

Also addressed the feedback by adding a byline. (And if you look at the PR history, you'll see that I got the co-authorship wrong at first and amended it. :sweat_smile: )
